physionet-django/project/cloud/s3.py

```diff
@@ -0,0 +1,749 @@
+    further to meet specific access requirements for the bucket.
+    - If 'public' is True, the policy allows public access to the
+    bucket. If 'public' is False, it restricts access to the specified
+    AWS accounts and users.
+    """
+    user = None
+    principal_value = (
+        "*"
+        if public
+        else {
+            "AWS": [
+                f"arn:aws:iam::{aws_id}:root"
+                if user is None or user == ""
+                else f"arn:aws:iam::{aws_id}:user/{user}"
+                for aws_id in aws_ids
+            ]
+        }
+    )
+
+    bucket_policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "AllowReadOnlyAccess",
+                "Effect": "Allow",
+                "Principal": principal_value,
+                "Action": ["s3:GetObject", "s3:ListBucket"],
+                "Resource": [
+                    f"arn:aws:s3:::{bucket_name}",
+                    f"arn:aws:s3:::{bucket_name}/*",
+                ]
+            }
+        ]
+    }
+
+    # Convert the policy from JSON dict to string
+    bucket_policy_str = json.dumps(bucket_policy)
+
+    return bucket_policy_str
+
+
+def set_bucket_policy(bucket_name, bucket_policy):
+    """
+    Apply a custom AWS S3 bucket policy to a specified bucket.
+
+    This function utilizes an AWS S3 client to set the custom
+    'bucket_policy' for the specified 'bucket_name', effectively
+    configuring access controls and permissions for objects
+    within the bucket.
+
+    Args:
+        bucket_name (str): The name of the AWS S3 bucket for
+        which to set the custom policy.
+        bucket_policy (str): A JSON-formatted string representing
+        the custom bucket policy.
+
+    Returns:
+        None
+
+    Note:
+    - Ensure that AWS credentials (Access Key and Secret Key)
+    are properly configured for the S3 client used in
+    this function.
+    - The 'bucket_policy' should be a valid JSON string adhering
+    to AWS S3 policy syntax.
+    """
+    s3 = create_s3_client()
+    # Check if s3 is None
+    if s3 is None:
+        return
+    s3.put_bucket_policy(Bucket=bucket_name, Policy=bucket_policy)
+
+
+def put_public_access_block(client, bucket_name, configuration):
+    """
+    Configure Amazon S3 Block Public Access settings
+    for a bucket.
+
+    Amazon S3 Block Public Access provides settings for buckets
+    to control and manage public access to Amazon S3 resources.
+    By default, new buckets do not allow public access.
+    This function sets the specified configuration for the given
+    bucket to allow or restrict public access.
+
+    Args:
+        client (boto3.client.S3): An initialized AWS S3
+        client object.
+        bucket_name (str): The name of the S3 bucket
+        to configure.
+        configuration (bool): The desired configuration for
+        public access. Set to 'True' to allow public access
+        or 'False' to restrict public access.
+
```

Comment:
I believe this is incorrect.  Please check.  Please rename the variable so that its meaning is clear:

If a `True` value means "forbid public access", then please rename `configuration` to `forbid_public_access`.

If a `True` value means "allow public access", then please rename `configuration` to `allow_public_access`.  (If that is the case, then the code in `update_bucket_policy` is wrong.)